src/d3a/models/market/two_sided_pay_as_bid.py

```diff
@@ -51,8 +51,8 @@ def __repr__(self):  # pragma: no cover
     def bid(self, price: float, energy: float, buyer: str, seller: str, bid_id: str=None) -> Bid:
         if energy <= 0:
             raise InvalidBid()
-
-        price = price * (1 - self.transfer_fee_ratio) - self.transfer_fee_const * energy
+        if price != 0:
```

Comment:
In this case, IMO we need to not create the new bid (and potentially warn the user with a warning message). The reason for that is that even if this bid is accepted, the grid fees would not be able to be paid.

Reply:
Yes you are right! How about something like this:
```
If price < self.transfer_fee_const * energy :
     raise InvalidBid()
```
... or some other Expection that tells the user to set the initial_buy_rate higher?

Reply:
I agree with the exception, I do not agree though on blowing it up to the user, because this is a perfectly valid use case to my eyes. I would instead try/catch for this exception in the forward_bid method, and in case the exception is raised the forwarding should be cancelled (potentially with a warning message).